src/pages/ListPage.tsx

```diff
@@ -1,28 +1,81 @@
 import { useHistory } from 'react-router-dom'
-import { Character, CharacterGender, CharacterStatus } from '../api/types'
+import { Character } from '../api/types'
 import { CharacterCard } from '../components/CharacterCard'
+import { FilterCard } from 'components/FilterCard'
+import { useHttp } from 'hooks/http.hook'
+import { useQuery } from 'react-query'
+import { useAppSelector } from 'hooks/useAppSelector'
+import { MouseEvent } from 'react'
+import { Alert, AlertTitle, Box, Button, Skeleton } from '@mui/material'
+import { useTypeDispatch } from 'hooks/useTypeDispatch'
+import { changeStatusActiveFilter, changeGenderActiveFilter } from 'store/reducers/filter'
+
+type requestType = {
+  info: {
+    count: number
+    pages: number
+    next: string
+    prev: string
+  }
+  results: Character[]
+}
 
 export const ListPage = () => {
+  const { activeStatus, activeGender, status, gender } = useAppSelector(state => state.filter)
+  const dispatch = useTypeDispatch()
+  const { request } = useHttp()
+
+  const fetchCharacters = async () => {
+    let query
+    if (activeStatus === 'All' && activeGender === 'All') {
+      query = ''
+    } else if (activeStatus === 'All' && activeGender !== 'All') {
+      query = `?gender=${activeGender}`
+    } else if (activeStatus !== 'All' && activeGender === 'All') {
+      query = `?status=${activeStatus}`
+    } else {
+      query = `?&status=${activeStatus}&gender=${activeGender}`
+    }
+    const data: requestType = await request(`https://rickandmortyapi.com/api/character${query}`)
+    return data
+  }
   const history = useHistory()
 
-  const handleCharacterClick = (id: number) => {
-    history.push(`/detail/${id}`)
+  const { isLoading, isIdle, error, data, refetch } = useQuery<requestType, Error>('characters', fetchCharacters)

```

Comment:
Zajímavý přístup se statickým queryKey a voláním refetch, není to špatně a funguje to ok, jen sem pro info napíšu co byla naše představa:

1. změna filtru se uloží v lokálním state (např. useState přímo ve filter komponentě).
2. kliknutím na tlačítko Apply se perzistují aktuální stavy filtrů do reduxu.
3. query key je dynamická (např: ['characters', status, gender]) takže není potřeba volat refetch, tím že se změní filtry v reduxu a změní se queryKey se provede refetch automaticky.

důsledky:
- rerender ListPage se provede jen 1x v případě aplikování filtrů. V tvém případě se rerenderuje s každou změnou filtru i celá ListPage se sezname charakterů.
- pro zachycení loading stavu musíš ve svém případě použít isFetching místo isLoading (isLoading pouze pokud jsou data undefined, isIdle jen pokud používáš enabled option, tady jsem ale smysl použití nevyčetl). Aktuálně tedy zobrazuješ skeleton jen při prvním loadingu (asi ok, pokud si to tak chtěl).